api/db/populate-attendants.js

```diff
@@ -0,0 +1,81 @@
+const { getRandomSubset, createPrompt } = require('./utils');
+
+const User = require('../models/User');
+const Event = require('../models/Event');
+const Attendant = require('../models/Attendant');
+
+let displayLogs;
+
+const createAttendant = async (attendantParams) => {
+  const attendant = new Attendant({
+    event: attendantParams.event,
+    user: attendantParams.user,
+  });
+  await attendant.save();
+
+  if (displayLogs) {
+    console.log('\n', '\x1b[0m', `New attendant created: ${attendant}`);
+  }
+};
+
+const createAttendants = async (verbose) => {
+  console.log('\n', '\x1b[0m', 'Populating attendants collection...');
+
+  let usersList = await User.distinct('_id');
+  if (usersList.length === 0) {
+    console.log(
+      '\x1b[33m',
+      'The users collection does not exist. Aborted attendants collection population.'
+    );
+    return;
+  }
+
+  let eventsList = await Event.find({
+    startDate: { $gte: new Date() },
+  });
+  if (eventsList.length === 0) {
+    console.log(
+      '\x1b[33m',
+      'The events collection does not exist. Aborted attendants collection population.'
+    );
+    return;
+  }
+
+  displayLogs = verbose;
+  let attendantsList = await Attendant.find();
+  const attendantsArray = [];
+
+  let affirmativeResponse = false;
+  if (attendantsList.length !== 0) {
+    affirmativeResponse = await createPrompt(
+      'Some events already contain attendees. Would you like to add attendees to events that already contain attendees?'
+    );
+  }
+
+  for (const event of eventsList) {
+    const attendees = attendantsList
+      .filter((attendee) => attendee.event.toString() == event._id.toString())
+      .map((attendee) => attendee.user.toString());
+
+    const availability = event.attendantsLimit - attendees.length;
+    if (!affirmativeResponse && event.attendantsLimit - 1 >= availability) {
```

Comment:
I fought with this one for a while. 😂 

I agree with what your saying on the latter point. However, it is such a convoluted  possibility that it is almost impractical to include when just populating attendees list. The attendees would essentially have to be populated, then a user would have to manually go in and unattend an event and the event would have to not have any other attendees, and then they would have to re-run the populate attendants script again. 

To handle such a situation would be expensive too. Plus ideally, once there are actual users, we wouldn't need the population scripts as much because you would probably want to have test data based off your actual production db.

Reply:
That being said, I can try to figure it out. 😅

Reply:
I figured it out. And of course it was super simple! 😅 Just had to take a break and come back to it.

Reply:
Hahaha, that's what normally happens! 🤣